client/components/data/purchases/edit-card-details/index.jsx

```diff
@@ -7,39 +7,38 @@ import React from 'react';
  * Internal dependencies
  */
 import CountriesList from 'lib/countries-list';
-import { fetchStoredCards, fetchUserPurchases } from 'lib/upgrades/actions';
+import { fetchUserPurchases } from 'lib/upgrades/actions';
 import observe from 'lib/mixins/data-observe';
 import PurchasesStore from 'lib/purchases/store';
+import QueryStoredCards from 'components/data/query-stored-cards';
 import { shouldFetchUserPurchases } from 'lib/purchases';
 import StoreConnection from 'components/data/store-connection';
-import StoredCardsStore from 'lib/purchases/stored-cards/store';
 import userFactory from 'lib/user';
 
 /**
  * Module variables
  */
 const stores = [
-		PurchasesStore,
-		StoredCardsStore
+		PurchasesStore
 	],
 	user = userFactory();
 
 function getStateFromStores( props ) {
 	return {
-		card: StoredCardsStore.getByCardId( parseInt( props.cardId, 10 ) ),
-		isEditingSpecificCard: Boolean( props.cardId ),
+		cardId: props.cardId,
 		countriesList: CountriesList.forPayments(),
 		hasLoadedSites: props.hasLoadedSites,
+		isDataLoading,
 		selectedPurchase: PurchasesStore.getByPurchaseId( parseInt( props.purchaseId, 10 ) ),
 		selectedSite: props.selectedSite
 	};
 }
 
 function isDataLoading( state ) {
 	return (
-		state.card.isFetching ||
 		! state.selectedPurchase.hasLoadedUserPurchasesFromServer ||
-		! state.hasLoadedSites
+		! state.hasLoadedSites ||
+		state.isFetchingStoredCards
```

Comment:
I'm not super fan of referring here to a prop that is in fact set in the `EditCardDetails` component but managing it in this data component proved to be awkward (or maybe that's just me :) ).